crates/bevy_asset/src/handle.rs

```diff
@@ -249,6 +248,34 @@ impl<A: Asset> PartialEq for Handle<A> {
 
 impl<A: Asset> Eq for Handle<A> {}
 
+impl<A: Asset> From<Handle<A>> for AssetId<A> {
+    #[inline]
+    fn from(value: Handle<A>) -> Self {
+        value.id()
+    }
+}
+
+impl<A: Asset> From<&Handle<A>> for AssetId<A> {
```

Comment:
We can replace this and the above with a blanket impl using the `Borrow` trait.

Reply:
Same below of course :)

Reply:
I don't think we can, because it appears to conflict with the existing blanket implementation for `From`:

```rust
impl<H> From<H> for UntypedAssetId where H: Borrow<UntypedHandle> {
    #[inline]
    fn from(value: H) -> Self {
        value.id()
    }
}
```

```
error[E0119]: conflicting implementations of trait `From<handle::UntypedHandle>` for type `id::UntypedAssetId`
   --> crates\bevy_asset\src\handle.rs:439:1
    |
432 | impl From<UntypedHandle> for UntypedAssetId {
    | ------------------------------------------- first implementation here
...
439 | impl<H> From<H> for UntypedAssetId where H: Borrow<UntypedHandle> {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `id::UntypedAssetId`
```

I think this is one of those instances where [`impl` specialisation](https://rust-lang.github.io/rfcs/1210-impl-specialization.html) would be required.